opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/RestTemplateAutoConfigurationTest.java

```diff
@@ -0,0 +1,67 @@
+package io.opentracing.contrib.spring.web.autoconfig;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+
+import io.opentracing.mock.MockTracer;
+
+/**
+ * @author Pavol Loffay
+ */
+@SpringBootTest(
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+        classes = {RestTemplateAutoConfigurationTest.SpringConfiguration.class})
+@RunWith(SpringJUnit4ClassRunner.class)
+public class RestTemplateAutoConfigurationTest {
+
+    @Configuration
+    @EnableAutoConfiguration
+    public static class SpringConfiguration {
+        @Bean
+        public MockTracer tracer() {
+            return new MockTracer();
+        }
+
+        @Bean
+        @Qualifier("foo")
+        public RestTemplate restTemplateFoo(RestTemplateBuilder builder) {
+            return builder.build();
+        }
+
+        @Bean
+        @Qualifier("bar")
+        public RestTemplate restTemplateBar(RestTemplateBuilder builder) {
+            return builder.build();
+        }
+    }
+
+    @Autowired
+    private MockTracer mockTracer;
+
+    @Autowired
+    @Qualifier("bar")
+    private RestTemplate restTemplate;
+
+    @Test
+    public void testTracingRequest() {
+        {
+            try {
+                restTemplate.getForEntity("http://www.abcfoobar.bar/baz", String.class);
```

Comment:
You should really use `example.com` (RFC 2606). If you need a non existing host, then prefix it with `nonexisting.example.com`, for instance.